### PR TITLE
UPBGE: Try to fix a crash with navmesh and memory leaks

### PR DESCRIPTION
**Pull request by youle31**
It is to try to fix:
https://github.com/UPBGE/upbge/issues/1437

I see no more memory leaks reported in blenderplayer

### CODE REVIEW

File: source/gameengine/Ketsji/KX_NavMeshObject.cpp [hunk 1]
```diff
@@ -810,6 +817,19 @@ bool KX_NavMeshObject::BuildNavMesh()
     if (vertices) {
       delete[] vertices;
     }
+    if (dvertices) {
```
Comment (lordloki) [inline source/gameengine/Ketsji/KX_NavMeshObject.cpp:820]: There is another delete [ ] dvertices more down. The same occurs in the other checks.

**Review by lordloki — APPROVED** [review]
It worked well. Great fix!!

File: source/gameengine/Ketsji/KX_NavMeshObject.cpp (hunk 1 above)
Comment (youle31) [reply source/gameengine/Ketsji/KX_NavMeshObject.cpp:820]: Thanks! I'll fix that.

**Comment by youle31** [comment]
Thanks very much for review and tests!

**Pull request merged**